utils/compress/idbcompress.cpp

```diff
@@ -107,53 +108,58 @@ namespace compress
 {
 #ifndef SKIP_IDB_COMPRESSION
 
-IDBCompressInterface::IDBCompressInterface(unsigned int numUserPaddingBytes) :
+CompressInterface::CompressInterface(unsigned int numUserPaddingBytes) :
     fNumUserPaddingBytes(numUserPaddingBytes)
 { }
 
-IDBCompressInterface::~IDBCompressInterface()
-{ }
-
 /* V1 is really only available for decompression, we kill any DDL using V1 by hand.
  * Maybe should have a new api, isDecompressionAvail() ? Any request to compress
  * using V1 will silently be changed to V2.
 */
-bool IDBCompressInterface::isCompressionAvail(int compressionType) const
+/*static*/
+bool CompressInterface::isCompressionAvail(int compressionType)
 {
-    if ( (compressionType == 0) ||
-            (compressionType == 1) ||
-            (compressionType == 2) )
-        return true;
+    return ((compressionType == 0) || (compressionType == 1) ||
+            (compressionType == 2) || (compressionType == 3));
+}
 
-    return false;
+size_t CompressInterface::getMaxCompressedSizeGeneric(size_t inLen)
+{
+    return std::max(snappy::MaxCompressedLength(inLen),
+                    LZ4_COMPRESSBOUND(inLen)) +
+           HEADER_SIZE;
 }
 
 //------------------------------------------------------------------------------
 // Compress a block of data
 //------------------------------------------------------------------------------
-int IDBCompressInterface::compressBlock(const char* in,
-                                        const size_t   inLen,
-                                        unsigned char* out,
-                                        unsigned int&  outLen) const
+int CompressInterface::compressBlock(const char* in, const size_t inLen,
+                                     unsigned char* out, size_t& outLen) const
 {
     size_t snaplen = 0;
     utils::Hasher128 hasher;
 
     // loose input checking.
-    if (outLen < snappy::MaxCompressedLength(inLen) + HEADER_SIZE)
+    if (outLen < maxCompressedSize(inLen) + HEADER_SIZE)
```

Comment:
HEADER_SIZE is added twice. Is that how it's meant to be?

Reply:
Thanks!

Reply:
oh, sorry, forgot to update it.